drivers/gles3/storage/light_storage.cpp

```diff
@@ -350,7 +350,14 @@ AABB LightStorage::light_get_aabb(RID p_light) const {
 	switch (light->type) {
 		case RS::LIGHT_SPOT: {
 			float len = light->param[RS::LIGHT_PARAM_RANGE];
-			float size = Math::tan(Math::deg_to_rad(light->param[RS::LIGHT_PARAM_SPOT_ANGLE])) * len;
+			float angle = Math::deg_to_rad(light->param[RS::LIGHT_PARAM_SPOT_ANGLE]);
+
+			if (angle > Math::PI / 2.0) {
+				// Light casts backwards as well.
+				return AABB(Vector3(-1, -1, -1) * len, Vector3(2, 2, 2) * len);
+			}
```

Comment:
Small optimization (the compiler doesn't do it for us as we don't use `-ffast-math`):

```suggestion
			if (angle > Math::PI * 0.5) {
				// Light casts backwards as well.
				return AABB(Vector3(-1, -1, -1) * len, Vector3(2, 2, 2) * len);
			}
```